tutorials/juliacon2022.jl

```diff
@@ -0,0 +1,1644 @@
+- `ranks()` - if features are `Taxon`s, returns a vector of the ranks of those taxa (if they have them)
+"""
+
+# ╔═╡ 5c85e0d5-d0aa-4e32-8ecd-e2688fc51201
+features(comm)
+
+# ╔═╡ 67bc83af-a427-4380-81fb-dbdb14cd2439
+samples(comm)
+
+# ╔═╡ 58a2aa85-8cce-4986-bfa2-5284afda40e6
+featurenames(comm)
+
+# ╔═╡ de790428-9248-4ec3-8bb6-b8fd187483ec
+ranks(comm)
+
+# ╔═╡ 95011cf6-e81b-4a48-9fdc-388df2c9ed75
+md"""
+### ℹ️ Indexing
+
+There are a number of useful ways to index into a `CommunityProfile`.
+The first and most straightforward is using numbers -
+keep in mind that indexing in julia starts at 1,
+that rows are the first dimension, and columns are the second dimension.
+
+For example, the following will return the relative abundance of the 3rd row and 6th column:
+"""
+
+# ╔═╡ c98a4d00-e94d-47aa-aa1b-d91c8422ddee
+comm[4, 6]
+
+# ╔═╡ 44e657e2-01bf-4fea-a2c3-c9eb900a6214
+md"""
+One can also get slices using standard julia syntax.
+Note that when taking slices, you get back another `CommmunityProfile`.
+"""
+
+# ╔═╡ e1f01522-8544-47ca-94ba-0deb9acaba15
+comm[1:3, 2:4]
+
+# ╔═╡ d3c9d708-f556-4ef6-a34b-c7e461351af4
+md"""
+One huge benefit of a `CommunityProfile`
+is that we can also index using strings and regular expressions
+that match on the `name` field of `AbstractFeature`s (rows)
+and `MicrobiomeSample`s:
+"""
+
+# ╔═╡ 9f2b2856-4256-48dc-8f02-a45dbc898798
+comm[r"Baci", :]
+
+# ╔═╡ ad1cec80-6f26-4541-a6af-2363257f609d
+md"""
+### 🌳 Ecological metrics
+
+There are a couple of built-in functions for calculating
+things like alpha diversity of samples.
+For example, to caclutate the Gini-Simpson or Shannon indecies,
+use `ginisimpson()` or `shannon()` respectively, both of which
+return a 1 x N matrix with length equal to the number of samples you have:
+"""
+
+# ╔═╡ 8529552b-bb06-49bd-937a-0bf21059b144
+shannon(comm)
+
+# ╔═╡ 15a5f712-415b-41c4-a3b0-d236075a7aba
+md"""
+Alternatively, you can use the mutating versions of these functions
+(`ginisimpson!()` and `shannon!()`) to automatically add the result
+to the metadata for each sample.
+"""
+
+# ╔═╡ c641f018-5e61-407c-a651-7c7f7b114449
+ginisimpson!(comm);
+
+# ╔═╡ 24aa8793-5470-4122-acd7-92a29312689b
+shannon!(comm);
+
+# ╔═╡ f322ff0a-49e4-41a2-96d7-e73d5c1f29bd
+samples(comm)
+
+# ╔═╡ 351a621b-c782-45dd-b141-d248eb07e8e8
+md"""
+We can also get information about the presence and prevalence of indivudual features.
+The `present()` function takes a `CommunityProfile`
+and an optional minimum abundance to be considered present (default > 0)
+and returns a boolean matrix:
+"""
+
+# ╔═╡ 8b6bc20b-6cc6-4450-a8c3-9b58fdbda6d8
+present(comm, 1.0) # anything over 1%
+
+# ╔═╡ 1e48e79b-7242-473b-a730-0ea7c5313c0f
+md"""
+The `prevalence()` function calculates the fraction of samples
+that a given feature is found, also with an optional minimum abundance value:
+"""
+
+# ╔═╡ 0b80a1b8-b6c6-4663-987d-b4613536056b
+prevalence(comm, 1.0)
+
+# ╔═╡ 7fdf6e5c-471e-4170-9a7c-2c027ec02100
+md"""
```

Comment:
some of the code for the text sections under this heading are not hidden for me?